src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java

```
@@ -65,19 +65,38 @@ public static synchronized String updateSourceCredentials(BuildConfig buildConfi
     return id;
   }
 
+  // getCurrentToken returns the ServiceAccount token currently selected by the user. If it catches a NullPointerException,
+  // it will try 10 times to get the token. If it still fails, it punts the exception. A return value of empty string
+  // implies no token is configured.
   public static String getCurrentToken() {
-    OpenShiftToken token = CredentialsMatchers.firstOrNull(
-            CredentialsProvider.lookupCredentials(
-                    OpenShiftToken.class,
-                    Jenkins.getActiveInstance(),
-                    ACL.SYSTEM,
-                    Collections.<DomainRequirement> emptyList()
-            ),
-            CredentialsMatchers.withId(GlobalPluginConfiguration.get().getCredentialsId())
-            );
-
-    if( token != null ) {
-      return token.getToken();
+    int retryAttempts = 0;
+
+    while(retryAttempts < 10) {
+      try {
+        OpenShiftToken token = CredentialsMatchers.firstOrNull(
+                CredentialsProvider.lookupCredentials(
+                        OpenShiftToken.class,
+                        Jenkins.getActiveInstance(),
+                        ACL.SYSTEM,
+                        Collections.<DomainRequirement>emptyList()
+                ),
+                CredentialsMatchers.withId(GlobalPluginConfiguration.get().getCredentialsId())
+        );
+
+        if (token != null) {
+          return token.getToken();
+        }
+
+        return "";
+      } catch(NullPointerException npe) {
```

catch `Throwable` instead of `NullPointerException`